---
Create by: Vizuet cf
Title: 2.2 Genome - Introduction to Problems and Challenges in Bioinformatics
---
# 2.2 Genome
## Sequence analysis

Some of the earliest problems in genomics concerned how to measure similarity of DNA and protein sequences, either within a genome, or across the genomes of different individuals, or across the genomes of different species. 

DNA and proteins can be similar in terms of their ***function***, their ***structure*** or their linear sequence of nucleotides or amino acids. 

The fundamental assumption for DNA is that two DNA sequences that are similar probably share the same function, even if they occur in different parts of the genome or across two or more genomes. 

The fundamental assumption for proteins is that linear sequence determines shape which, in turn, determines function. This is because the shape of a protein, and in particular of enzymes, determines which other molecules these proteins can lock on to and affect.

Consider the two DNA strings of equal length: ACGTACGT and ACCTAGGT. How similar are they? One way to deal with this problem is to place them one on top of the other:

> ***Secuencias***:
> 
>   $A$ $C$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $C$ $G$ $T$  $A$ $G$ $G$ $T$  

> ***Hamming distance***
> 
> - A count is made column by column to identify the number of mismatches per position, which in the above case is two. This is the ***Hamming distance***, which is the simplest measure of similarity available.

The two strings $ACGTACGT$ and $CCCTCCCT$ would have a Hamming distance of four, and the two strings $ACCTAGGT$ and $CCCTCCCT$ would also have a Hamming distance of four. 

The two strings $ACGTACGT$ and $ACCTAGGT$ therefore are more similar to each other (Hamming distance of two) than $CCCTCCCT$ is to either of them (Hamming distance of four). 

What happens if strings are of unequal length? Consider ACGTACGT and $AGTACGT$. If these strands are lined up:

> ***Secuencias***:
> 
>   $A$ $C$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $G$ $T$ $A$  $C$ $G$ $T$  

The Hamming distance of seven (assuming that the last base of the first string cannot be matched to a blank). 

Yet, if a blank is inserted in the second string:

> ***Secuencias***:
> 
>   $A$ $C$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $-$ $G$ $T$ $A$  $C$ $G$ $T$  

The Hamming distance is one, i.e. the strings are very similar.

Now imagine that, instead of just eight bases in a DNA sequence there are hundreds and possibly thousands of bases (for example, if a whole gene is compared against other genes). 

Gene sequences are extremely unlikely to be of equal length, and methods must be found for inserting blanks at appropriate locations in the shorter string and stretching it out to optimize the number of matches. 

Shorter strings may result when the DNA replication machinery goes wrong and bases are skipped over.

Equally, some bases may need to be deleted. Consider the following three strings:

> ***Secuencias***:
> 
>   $A$ $C$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $G$ $G$  $A$ $C$ $G$ $T$  

One possibility is to insert blanks into the second and third strings at position two (two **insertions**) to line up the three strings. Another possibility is to delete the second base of the first string (one **deletion**):

> ***Secuencias***:
> 
>   $A$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $G$ $T$  $A$ $C$ $G$ $T$  
>   $A$ $G$ $G$  $A$ $C$ $G$ $T$  

Since one deletion may be preferable to two insertions this may be the preferred strategy, but now consider what would happen if the first two strings were matched without any knowledge of the third string. The strategy might well have been to insert a blank into position two of the second string to optimize similarity.

However, when the third string is entered, it is now discovered that it would have been preferable to delete the second base of the first string rather than insert a blank into the third string. 

Backtracking may be required to undo the insertion of the blank into the second string, but backtracking will only work if there is stored information as to what was done earlier so that it can be undone. For long strings and for matching many strings, the memory requirements can quickly become large.

The above problem is easy with just a handful of strings and small numbers of bases, but already the problem with long and large numbers of sequences is apparent. There can be ***pairwise comparison*** of strings, where changes are made to earlier decisions as new strings are entered, or there can be ***multiple comparison*** of all strings at once and matches can be optimized for specific positions across all sequences. Also, there can be ***local alignment*** (finding alignments between parts of two or more sequences) and ***global alignment*** (finding an alignment for sequences in their entirety). There are now a number of publicly available tools on the web for undertaking alignments.[1],[2]

The requirement for a minimal number of changes arises from the principle that, when identifying similarity between strings, as few alterations as possible should be made to the original strings so that optimal similarity measures are returned. This is the unit cost model, also known as the Levenshtein Distance, which states that the cost of an alignment of two sequences $s1$ and $s2$ is the sum of the costs of all the ‘**edit**’ operations required to match the two sequences, and that an ***optimal*** alignment of $s1$ and $s2$ is an alignment that has minimal cost among all the possible ways that they can be aligned. 

Extensions to the unit cost model include ***substitution matrices*** that provide variable costs for insertion, deletion and replacement of bases and amino acids, ***realistic gap models*** that prevent deletions and insertions in critical subsequences (such as strongly conserved subunits in protein sequences involved in protein–protein interaction, where any edit in these subsequences may destroy the desired biochemical function) and the use of an extended genetic alphabet that represents possible ambiguities in the data. 

The most common symbols used in an extended genetic alphabet are: $R$ for $G$ or $A$ (Pu$R$ine), $Y$ for $T$ or $C$ (P$Y$ramidine) and $N$ (A$N$y).

A related problem here is how to find a common substring for all strings or sequences. This is known as the ‘**superstring**’ problem, where the common substring is the shortest sequence of characters shared by all sequences. 

This problem is, in computational terms, ***intractable***, in that there is no known algorithm that will work in reasonable time to find such a superstring as the number of sequences and their length increase.



